vcd/config.go

```diff
@@ -90,6 +91,79 @@ func debugPrintf(format string, args ...interface{}) {
 	}
 }
 
+// This is a global MutexKV for all resources
+var vcdMutexKV = mutexkv.NewMutexKV()
+
+func (cli *VCDClient) lockVapp(d *schema.ResourceData) {
+	vappName := d.Get("name").(string)
+	if vappName == "" {
+		panic("vapp name isn't found")
+	}
+	key := fmt.Sprintf("org:%s|vdc:%s|vapp:%s", cli.getOrgName(d), cli.getVdcName(d), vappName)
+	vcdMutexKV.Lock(key)
+}
+
+func (cli *VCDClient) unLockVapp(d *schema.ResourceData) {
+	vappName := d.Get("name").(string)
+	if vappName == "" {
+		panic("vapp name isn't found")
+	}
+	key := fmt.Sprintf("org:%s|vdc:%s|vapp:%s", cli.getOrgName(d), cli.getVdcName(d), vappName)
+	vcdMutexKV.Unlock(key)
+}
+
+func (cli *VCDClient) lockParentVapp(d *schema.ResourceData) {
+	vappName := d.Get("vapp_name").(string)
+	if vappName == "" {
+		panic("vapp name isn't found")
+	}
+	key := fmt.Sprintf("org:%s|vdc:%s|vapp:%s", cli.getOrgName(d), cli.getVdcName(d), vappName)
+	vcdMutexKV.Lock(key)
+}
+
+func (cli *VCDClient) unLockParentVapp(d *schema.ResourceData) {
+	vappName := d.Get("vapp_name").(string)
+	if vappName == "" {
+		panic("vapp name isn't found")
+	}
+	key := fmt.Sprintf("org:%s|vdc:%s|vapp:%s", cli.getOrgName(d), cli.getVdcName(d), vappName)
+	vcdMutexKV.Unlock(key)
+}
+
+func (cli *VCDClient) lockParentEdgeGtw(d *schema.ResourceData) {
+	edgeGtwName := d.Get("edge_gateway").(string)
+	if edgeGtwName == "" {
+		panic("edge gtw name isn't found")
```

Comment:
"edge gtw name isn't found" -> "edge gateway not found"

Reply:
Fixed